### PR TITLE
Speciality model

### PR DESCRIPTION
**Pull request by principles31**
# The following were implemented in this **PR** as the project requirements: 
> Create Speciality model
> Set validations
> Define associations
> has many doctor's

### CODE REVIEW

File: app/models/speciality.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class Speciality < ApplicationRecord
+  validates :name, :description, presence: true
+  validates :rating, length: { maximum: 1 }, numericality: { less_than_or_equal_to: 5 }, allow_nil: true
+
+  has_many :doctors, dependent: :destroy
+end
```
Comment (Arturgouveia1970) [inline app/models/speciality.rb]: Kindly add the doctor's foreign key to the speciality table.

File: db/migrate/20221010922901_create_specialists.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+class CreateSpecialists < ActiveRecord::Migration[7.0]
+    def change
+      create_table :specialists do |t|
+        t.string :name, null: false, index: true
+        t.text :description, null: false
+        t.integer :rating, comment: 'Specialists Doctor rating system', limit: 1
```
Comment (Arturgouveia1970) [inline db/migrate/20221010922901_create_specialists.rb]: Kindly remove the rating as you cannot rate a doctor's specialy.

**Review by Arturgouveia1970 — CHANGES_REQUESTED** [review]
Hello @principles31, you have done a great job creating the specialities model and controller, however, I believe that you need to make a couple of changes:

**Comment by principles31** [comment]
- [x] Kindly add the doctor's foreign key to the speciality table.
- [x] Kindly remove the rating as you cannot rate a doctor's speciality.
- [x] The above changes requested by the reviewer have been implemented.

**Review by Arturgouveia1970 — APPROVED** [review]
Hello @principles31, Job well done, I can see that you have implemented all changes required by the last reviewer and that a speciality can be created. Congratulations, it's time to merge.

**Pull request merged**